### PR TITLE
fix: fix master config and experiment config for log retention

### PR DESCRIPTION
**Pull request by salonig23**
## Description

Fixing the experiment and master config to make it consistent and adding a wrapper layer.

## Test Plan

Make sure all tests pass.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/reference/experiment-config-reference.rst [hunk 1]
```diff
@@ -321,18 +321,26 @@ defaults.
 
 .. _log-retention-days:
 
-``log_retention_days``
-======================
+``retention_policy``
+====================
+
+Optional. Defines retention policies for logs of all the trials of a given experiment. Parameters
+include:
```
Comment (tara-det-ai) [inline docs/reference/experiment-config-reference.rst:328]: Optional. Defines retention policies for logs related to all trials of a given experiment. Parameters
include:

File: docs/reference/experiment-config-reference.rst [hunk 2]
```diff
@@ -321,18 +321,26 @@ defaults.
...
 
-Optional. Overrides the number of days to retain logs for a trial. Values should be between ``-1``
-and ``32767``. If set to ``-1``, logs will be retained indefinitely. If set to ``0``, logs will be
-deleted during the next cleanup.
+-  ``log_retention_days``: Optional. Overrides the number of days to retain logs for a trial. Values
+   should be between ``-1`` and ``32767``. If set to ``-1``, logs will be retained indefinitely. If
```
Comment (tara-det-ai) [inline docs/reference/experiment-config-reference.rst:331]: -  ``log_retention_days``: Optional. Overrides the number of days to retain logs for a trial. Acceptable values range from ``-1`` to ``32767``. If set to ``-1``, logs will be retained indefinitely. If

File: docs/reference/experiment-config-reference.rst [hunk 3]
```diff
@@ -321,18 +321,26 @@ defaults.
...
+   set to ``0``, logs will be deleted during the next cleanup. To change the retention settings
+   after completion for a single trial or the entire experiment, the CLI command ``det t set
```
Comment (tara-det-ai) [inline docs/reference/experiment-config-reference.rst:333]: set to ``0``, logs will be deleted during the next cleanup. To modify the retention settings post-completion for a single trial or the entire experiment, you can use the CLI command

File: docs/reference/experiment-config-reference.rst [hunk 4]
```diff
@@ -321,18 +321,26 @@ defaults.
...
+   log-retention <trial-id>`` or ``det e set log-retention <exp-id>`` can be used. Both of these
+   commands accept two arguments :``--days`` which sets the number of days to retain the logs for,
+   from the time of creation or ``--forever`` which retains logs forever.
```
Comment (tara-det-ai) [inline docs/reference/experiment-config-reference.rst:336]: Both commands accept either the argument :``--days``, which sets the number of days to retain logs from the time of creation, or ``--forever`` which retains logs indefinitely.

**Review by tara-det-ai — APPROVED** [review]
suggested edits

**Review by jgongd — APPROVED** [review]
model dev stamp

**Pull request merged**